common_types/src/schema.rs

```diff
@@ -206,11 +207,48 @@ pub enum CompatError {
 /// Meta data of the arrow schema
 #[derive(Default)]
 pub struct ArrowSchemaMeta {
-    num_key_columns: usize,
+    primary_key_indexes: Indexes,
     timestamp_index: usize,
     version: u32,
 }
 
+#[derive(Debug, Default, PartialEq)]
+struct Indexes {
+    pub indexes: Vec<usize>,
+}
+
+impl ToString for Indexes {
+    fn to_string(&self) -> String {
+        format!("{:?}", self.indexes)
```

Comment:
We shouldn't rely on `debug format` to format indexes, what if it changed between different Rust version?
```
self.indexes.join(',')
```

This is enough.